tests/pyfunc/test_pyfunc_schema_enforcement.py

```diff
@@ -12,6 +12,23 @@
 import pytest
 import sklearn.linear_model
 from packaging.version import Version
+from pyspark.sql import SparkSession
```

Comment:
We might want to split this out as its own test suite and adjust CI so that we're not running the entire group of python core tests in an environment that has pyspark installed. cc @harupy for thoughts and opinions here.